src/lib.rs

```diff
@@ -53,6 +56,12 @@ fn get_internal() -> usize {
     unsafe { kernel32::GetCurrentThreadId() as usize }
 }
 
+#[cfg(target_os = "redox")]
+#[inline]
+fn get_internal() -> usize {
+    syscall::getpid().unwrap()
```

Comment:
It is true that threads are simply processes on Redox, which share memory with other processes.